recipes/material-ui/index.ts

```diff
@@ -0,0 +1,443 @@
+import {RecipeBuilder, paths, addImport} from "@blitzjs/installer"
+import {builders} from "ast-types/gen/builders"
+import {ASTNode} from "ast-types/lib/types"
+import {NamedTypes, namedTypes} from "ast-types/gen/namedTypes"
+import {visit} from "ast-types"
+
+export default RecipeBuilder()
+  .setName("Material-UI")
+  .setDescription(
+    `Configure your Blitz app's styling with Material-UI. This recipe will install all necessary dependencies and configure a base Material-UI setup for immediate usage.
+    
+    NOTE: Material-UI currently doesn't support concurrent mode. Therefore, if you intend to use it with blitz.js, you will need to set { suspense: false } on useQuery hook for fetching data.`,
+  )
+  .setOwner("s.pathak5995@gmail.com")
+  .setRepoLink("https://github.com/blitz-js/blitz")
+  .addAddDependenciesStep({
+    stepId: "addDeps",
+    stepName: "Add npm dependencies",
+    explanation: `@material-ui/core needs to be installed`,
+    packages: [{name: "@material-ui/core", version: "latest"}],
+  })
+  .addTransformFilesStep({
+    stepId: "modifyGetInitialPropsInCustomDocumentApp",
+    stepName: "Add custom getInitialProps logic in Custom Document",
+    explanation: `We will add custom getInitialProps logic in _document. We need to do this so that styles are correctly rendered on the server side.`,
+    singleFileSearch: paths.document(),
+    transform(ast: ASTNode, b: builders, t: NamedTypes) {
+      if (t.File.check(ast)) {
+        // import ServerStyleSheets
+        const serverStyleSheetsImport = b.importDeclaration(
+          [b.importSpecifier(b.identifier("ServerStyleSheets"))],
+          b.literal("@material-ui/core/styles"),
+        )
+
+        let isReactImported = false
+
+        visit(ast, {
```

Comment:
I'm gonna update the `addImport` function we currently expose to take just a module name and a `ImportSpecifier | ImportDefaultSpecifier` so we can hide all of this logic away. It's fine to merge this as-is if we want, but otherwise we can remove a lot of this boilerplate